enhancement: JSON variable file support

sunday2 opened the pull request:
Closes #4532

pekkaklarck reviewed (changes requested):
Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

sunday2 commented:
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

sunday2 commented:
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

sunday2 commented on src/robot/variables/filesetter.py:
```diff
@@ -147,3 +150,27 @@ def _validate(self, name, value):
+                for name, value in variables]
+
+    def _import(self, path):
+        with io.open(path, encoding='UTF-8') as stream:
```
Thanks for the explain and I will not change both of them in this PR.

sunday2 commented:
> 

Thank you for the review and all the comments, I have updated the code. If there is any other concern, pls don't hesitate to let me know. For writing document, maybe I need to spend some time on studying how to write it.

sunday2 commented:
> Thanks for a great PR and sorry for reviewing taking so long. Only real issue I noticed in code and tests was that the extension wasn't case-insensitive as it should be, but I added also some other comments. Another thing to do is writing documentation. It can be done after merging this PR as well.

Thank you for the review and all the comments, I have updated the code. If there is any other concern, pls don't hesitate to let me know. For writing document, maybe I need to spend some time on studying how to write it.

![image](https://user-images.githubusercontent.com/20227068/222938894-15ea948f-3e84-4947-85c1-1b58d3acd282.png)

pekkaklarck commented:
Thanks for a great PR! I can write the documentation myself, I think it can mainly refer to existing YAML variable file documentation so it's a small task.
